nilearn/datasets/utils.py

```diff
@@ -774,6 +766,85 @@ def _fetch_files(data_dir, files, resume=True, mock=False, verbose=1):
     return files_
 
 
+def _message_or_mock_missing_dataset_file(file_, mock, abort, dl_file, target_file,
+                                          temp_target_file, url):
+    """ Warns & returns an abort message if the downloaded dataset does not have the intended file.
+    If mock==True (intended for testing), creates a mock file instead.
+    Returns
+    -------
+    abort: None or String
+    """
+    
+    if mock:
+        if not os.path.exists(os.path.dirname(temp_target_file)):
+            os.makedirs(os.path.dirname(temp_target_file))
+        open(temp_target_file, 'w').close()
+    else:
+        warnings.warn('An error occured while fetching %s' % file_)
+        abort = ("Dataset has been downloaded but requested file was "
+                 "not provided:\nURL: %s\n"
+                 "Target file: %s\nDownloaded: %s" %
+                 (url, target_file, dl_file))
+    return abort
+
+
+def _check_files_existence(file_1, file_2):
+    """ Returns True if both the provided files exist.
+    """
+    target_file_does_not_exist = not os.path.exists(file_1)
+    temp_target_file_does_not_exist = not os.path.exists(file_2)
+    file_does_not_exist = target_file_does_not_exist and temp_target_file_does_not_exist
+    return file_does_not_exist
+    
+
+def _uncompress_dataset_files(file_, mock, abort, verbose):
+    """ Uncompresses a file. Removes it if its size is zero or mock=True
+    
+    Returns
+    -------
+    
+    None or String(exception)
+    """
+    try:
+        if not mock or os.path.getsize(file_) != 0:
+            _uncompress_file(file_, verbose=verbose)
+        else:
+            os.remove(file_)
+    except Exception as e:
+        abort = str(e)
+    return abort
```

Comment:
This function is used only once, and understanding what it does is as difficult from the code as from the function name and description. It should not be factored out.